cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/runtimejob/DataprocRuntimeJobManager.java

```diff
@@ -234,9 +248,12 @@ public void launch(RuntimeJobInfo runtimeJobInfo) throws Exception {
       for (LocalFile fileToUpload : localFiles) {
         boolean isCacheable = !disableGCSCaching && fileToUpload instanceof CacheableLocalFile;
         String targetFilePath = getPath(isCacheable ? cacheRootPath : runRootPath, fileToUpload.getName());
+        String targetFilePathWithVersion = getPath(cacheRootPath, appCDAPVersion[0], fileToUpload.getName());
         uploadFutures.add(
           provisionerContext.execute(() -> isCacheable ? uploadCacheableFile(bucket, targetFilePath, fileToUpload) :
-              uploadFile(bucket, targetFilePath, fileToUpload, false))
+              commonArtifactsPerCDAPVersion.contains(fileToUpload.getName()) && enableCommonArtifactsCaching ?
```

Comment:
this inline if statement has become hard to read. I believe it'll be cleaner if we use normal if block.

Reply:
ack